tests/integration/test_client.py

```diff
@@ -589,6 +590,124 @@ async def test_batch(client: Client, table: TableName) -> None:
     assert len([item async for item in client.query(table, HashKey("h", "h"))]) == 0
 
 
+async def test_transact_write_items(
+    client: Client, table: TableName, is_dynalite_emulator: bool
+) -> None:
+    if is_dynalite_emulator:
+        pytest.xfail("transaction not supported by dynalite yet")
+
+    with pytest.raises(errors.TransactionCanceled):
+        await client.transact_write_items(items=[])
+
+    with pytest.raises(errors.TransactionCanceled):
+        puts = [Put(table=table, item={"h": "h", "r": str(i)}) for i in range(26)]
+        await client.transact_write_items(items=puts)
```

Comment:
I think this test shows that we need to handle `TransactionCanceled` in a better way. The [documentation](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_TransactWriteItems.html#API_TransactWriteItems_Errors) shows that that error has a nested error which hints at the actual issue. I think we should capture that inner error and raise that instead.